### PR TITLE
Splitting sensitive files rules

### PR DESCRIPTION
**Pull request by eddynaka**

### CODE REVIEW

File: Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json [hunk 1]
```diff
@@ -4,16 +4,111 @@
...
       "Id": "SEC1003",
       "Level": "Error",
-      "FileNameAllowRegex": "(?i)\\.(pem|pkcs12|p12|pfx|asc|ovpn|rdp|mdf|sdf|sqlite|sqlite3|bek|tpm|fve|jks|psafe3|agilekeychain|kwallet|tblk|ppk)$",
+      "FileNameAllowRegex": "(?i)\\.(agilekeychain|asc|bek|fve|jks|kwallet|mdf|ovpn|p12|pem|pfx|pkcs12|ppk|psafe3|rdp|sdf|sqlite|sqlite3|tblk|tpm)$",
       "Description": "Review potentially sensitive files in versioned engineering content and other locations for security risk.",
       "Message": "'{0:scanTarget}' is {1:fileKind} that may compromise security if visible to attackers.",
-
       "MatchExpressions": [
+        {
+          "SubId": "1PasswordFile",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json:12]: >1PasswordFile [](start = 20, length = 13)

Uh-oh! Our validator convention will look for {SubId}Validator as a type name and C# types can't be prefixed with a number. :) So it won't be possible to author a validator for this one currently. We can worry about this later. :)

File: Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json [hunk 2]
```diff
@@ -4,16 +4,111 @@
...
+          "FileNameAllowRegex": "(?i)\\.agilekeychain$",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json:13]: >agilekeychain$ [](start = 40, length = 14)

we should also flag .agilekeychain_zip files, apparently 1password supports compressing these and this is the naming convention

File: Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json [hunk 3]
```diff
@@ -4,16 +4,111 @@
...
+          "MessageArguments": { "fileKind": "a Java keystore file" }
+        },
+        {
+          "SubId": "KDEWalletManagerDatabaseFile",
+          "FileNameAllowRegex": "(?i)\\.kwallet$",
+          "ContentsRegex": ".",
+          "MessageArguments": { "fileKind": "a KDE Wallet Manager database file" }
+        },
+        {
+          "SubId": "MicrosoftSQLDatabaseFile",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json:48]: >MicrosoftSQLDatabaseFile [](start = 20, length = 24)

For this name use MicrosoftSqlDatabaseFile. Note the casing of Sql...

File: Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json [hunk 4]
```diff
@@ -4,16 +4,111 @@
...
+          "FileNameAllowRegex": "(?i)\\.mdf$",
+          "ContentsRegex": ".",
+          "MessageArguments": { "fileKind": "a Microsoft SQL database file" }
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json:51]: >SQL [](start = 57, length = 3)

The SQL casing is fine here! 'Sql' is preferred for API-style things, like an ID or class name. SQL is appropriate casing for user-facing strings.

File: Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json [hunk 5]
```diff
@@ -4,16 +4,111 @@
...
+          "SubId": "OpenVPNClientConfigurationFile",
+          "FileNameAllowRegex": "(?i)\\.ovpn$",
+          "ContentsRegex": ".",
+          "MessageArguments": { "fileKind": "an OpenVPN client configuration file" }
+        },
+        {
+          "SubId": "P12CryptographicKeyFile",
+          "FileNameAllowRegex": "(?i)\\.p12$",
+          "ContentsRegex": ".",
+          "MessageArguments": { "fileKind": "a cryptographic key bundle" }
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json:63]: >cryptographic [](start = 47, length = 13)

say 'a PKCS#12 cryptographic key bundle? This is the same description for PFX files, maybe we merge the rules? Add PKCS12 as well? I think these are all files driven by the same container spec, PKCS#12

File: Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json (hunk 1 above)
Comment (eddynaka) [reply Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json:12]: I can change to OnePasswordFile. thats fine.

---
In reply to: [545472151](https://github.com/microsoft/sarif-pattern-matcher/pull/39#discussion_r545472151) [](ancestors = 545472151)

File: Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json [hunk 6]
```diff
@@ -4,16 +4,111 @@
...
+        },
+        {
+          "SubId": "PEMCryptographicPrivateKeyFile",
+          "FileNameAllowRegex": "(?i)\\.pem$",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json:67]: >pem$ [](start = 40, length = 4)

a PEM file is *not* a PKCS#12 so it should be its own rule.  use 'a PEM cryptographic file' as a description. Strictly speaking, you can emit a PEM with only the public key (which is not a security risk). Let's open a tracking item for a validator that looks for this condition.

File: Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json [hunk 7]
```diff
@@ -4,16 +4,111 @@
...
+          "MessageArguments": { "fileKind": "a cryptographic private key" }
+        },
+        {
+          "SubId": "PFXCryptographicKeyFile",
+          "FileNameAllowRegex": "(?i)\\.pfx$",
+          "ContentsRegex": ".",
+          "MessageArguments": { "fileKind": "a cryptographic key bundle" }
+        },
+        {
+          "SubId": "PKCS12CryptographicKeyFile",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json:78]: >PKCS12CryptographicKeyFile [](start = 20, length = 26)

btw - this is probably the right SubId for all these PKCS12 container extensions when you create the roll-up rule.

File: Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json [hunk 8]
```diff
@@ -4,16 +4,111 @@
...
+          "FileNameAllowRegex": "(?i)\\.pkcs12$",
+          "ContentsRegex": ".",
+          "MessageArguments": { "fileKind": "a cryptographic key bundle" }
+        },
+        {
+          "SubId": "PuTTygenPrivateKeyFile",
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json:84]: >PuTTygenPrivateKeyFile [](start = 20, length = 22)

Nice work using the right casing here. :) We adhere to specialized casing for external companies in many cases. btw - if our rule *says* we are detecting a private key key, we want to be sure we flag only containers that hold private keys (and are not restricted to public). maybe have a bucket issue for chasing all this down, it'd be a good focus topic at some point.

File: Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json [hunk 9]
```diff
@@ -4,16 +4,111 @@
...
+          "FileNameAllowRegex": "(?i)\\.ppk$",
+          "ContentsRegex": ".",
+          "MessageArguments": { "fileKind": "a PuTTYgen private key" }
+        },
+        {
+          "SubId": "PasswordSafeDatabaseFile",
+          "FileNameAllowRegex": "(?i)\\.psafe3$",
+          "ContentsRegex": ".",
+          "MessageArguments": { "fileKind": "a password Safe database file" }
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json:93]: >password [](start = 47, length = 8)

I'd say something like so here. Note that I'm encapsulating the company name in apostrophes to make it clear this is the company name:

a 'Password Safe' database file

File: Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json [hunk 10]
```diff
@@ -4,16 +4,111 @@
...
+        },
+        {
+          "SubId": "RemoteDesktopConnectionFile",
+          "FileNameAllowRegex": "(?i)\\.rdp$",
+          "ContentsRegex": ".",
+          "MessageArguments": { "fileKind": "a Remote Desktop connection file" }
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SEC1003.ReviewPotentiallySensitiveFiles.json:99]: >Remote Desktop connection [](start = 47, length = 25)

let's say 'a Remote Desktop connection settings file'

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**